services/security/authentication/casigned/casigned.go

```diff
@@ -61,9 +109,33 @@ func (c *client) LoginWithConfig(group string, loginconfig auth.LoginConfig) (*a
 	accessFile.ClientCertificateType = auth.CASigned
 	accessFile.IdentityName = loginconfig.Name
 	auth.PrintAccessFile(accessFile)
+	go RenewRoutine(group, c.cloudFQDN)
```

Comment:
How do we prevent multiple goroutines from happening? Do we need a lock here?

Reply:
Good point. multiple logins could trigger multiple go routines.

Reply:
go routine should be triggered only if we created a certificate. we could exit before creating a certificate, if it already exists

Reply:
We do not know the state of the old certificate that exists